site/content/en/docs/Installation/upgrading.md

```diff
@@ -128,7 +128,97 @@ upgrades.
    between the Agones controller being recreated and GameServers being deleted doesn't occur, and GameServers can end up stuck in erroneous states.
 1. Start and complete your control plane upgrade(s).
 1. Start and complete your node upgrades.
-1. Scale your Fleets back up and/or recreate your GameServers. 
+1. Scale your Fleets back up and/or recreate your GameServers.
 1. Run any other tests to ensure the Agones installation is still working as expected.
 1. Close your maintenance window.
 7. Congratulations - you have now upgraded to a new version of Kubernetes! 👍
+
+{{% feature publishVersion="1.41.0" %}}
+## SDK Compatibility Guarantees
+
+The SDK compatibility contract aims to ensure smooth upgrades and reduce the need for frequent
+binary redeployments for game server users.
+
+What’s the point of having a compatibility guarantee?
+Goal: Reduce user toil by minimizing the need for frequent binary redeployments due to SDK changes.
+Versioning: Clear documentation of API maturity levels and history (Stable, Beta, Alpha) with
+release information.
+
+Our SDK Server compatibility contract as of Agones v1.41.0: A
+game server binary using Beta and Stable SDKs will remain compatible with a _newer_ Agones Release,
+within possible deprecation windows:
+- If your game server uses a non-deprecated Stable API, your binary will be compatible for 10
+releases (~1y) or more starting from the SDK version packaged.
+  - For example, if the game server uses non-deprecated stable APIs in the 1.40 SDK, it will be
+  compatible through the 1.50 SDK.
+  - Stable APIs will almost certainly be compatible beyond 10 releases, as deprecation of stable
+  APIs is rare, but 10 releases is guaranteed.
+- If your game server uses a non-deprecated Beta API, your binary will be compatible for 5 releases
+(~6mo).
+- Alpha SDK APIs are subject to change between releases.
+  - A game server binary using Alpha SDKs may not be compatible with a newer sdk-server.
+  - In Alpha, incompatible changes retaining the same SDK proto message name are allowed.
```

Comment:
```suggestion
  - In Alpha, incompatible SDK changes retaining the same SDK proto message formats will continue to work.
```

Reply:
This sentence has a slightly different meaning. Generally with proto messages a field is immutable -- if you want to make a change to a field you have to deprecate the field and create a new field with a different name. This is saying we allow changes to existing fields for Alpha only protos.

Reply:
oh yeah, I get you!

Since we say "  - A game server binary using Alpha SDKs may not be compatible with a newer Agones release if
  breaking changes have been made between releases." - maybe we drop this line. I don't think we need to document how it could break? WDYT?

Reply:
We can document in the release notes when breaks happen (which we say we will do).